src/usdb_syncer/notes_parser.py

```diff
@@ -300,6 +303,28 @@ def fix_spaces(self) -> None:
             # used to be incomplete in USDX, and it allows simple text concatenation
             line.notes[-1].right_trim_text_and_add_space()
 
+    def fix_all_caps(self) -> None:
+        if self.lyrics().isupper():
```

Comment:
You don't need to construct a new string here:
```suggestion
        if all(note.text.isupper() for note in self.all_notes()):
```